src/components/GarminFilePicker/GarminFilePickerContainer.tsx

```diff
@@ -24,8 +27,26 @@ export function GarminFilePickerContainer() {
     const fileList = Array.from(files);
 
     // send files to context
-    uploadActivities(fileList);
+    // this can fail if local storage is full
+    // TODO: make local storage evict keys
```

Comment:
I tried to think about implementing a JSON-based LRU cache and then decided I value my sanity